src/transactions/ManageOfferOpFrame.cpp

```diff
@@ -164,6 +167,8 @@ ManageOfferOpFrame::doApply(Application& app, LedgerDelta& delta,
             return false;
         }
 
+        mSellSheepOffer->storeDelete(tempDelta, db);
```

Comment:
I think you should also make the call to `addNumEntries` here to keep the account in a consistent state (and behavior is basically the same between modify and create except for result).

I think it just means that the condition later needs to be updated to
```
if (!creatingNewOffer ||
                 app.getLedgerManager().getCurrentLedgerVersion() > 8)
```